### ISSUE TITLE
bug: Components start loading before globalscript completes

### ISSUE THREAD

**Issue by runr02** [ionic-team/stencil#4889]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

9.6.4

### Current Behavior

Components start loading before globalscript completes. 

### Expected Behavior

Stencil documentation states the globalscript runs once before the library/app loads. As I understand this, the globalscript should complete before other components render. 

### System Info

```shell
System: node 18.16.0
    Platform: windows (10.0.19045)
       Build: 1696258196
     Stencil: 4.4.0
  TypeScript: 5.2.2
      Rollup: 2.42.3
      Parse5: 7.1.2
      Sizzle: 2.42.3
      Terser: 5.20.0
```

### Steps to Reproduce

See repo below. 

### Code Reproduction URL

https://github.com/runr02/globalScriptAsyncIssue

**Comment by rwaskiewicz** [comment]
Thanks! I appreciate the time to put together a minimal reproduction case. I've confirmed that this is a bug with the reproduction case. However, I'm going to close this as a duplicate of https://github.com/ionic-team/stencil/issues/3392. Thanks again!
